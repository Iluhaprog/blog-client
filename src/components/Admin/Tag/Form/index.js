import {TagForm} from './TagForm';
export {TagForm};
