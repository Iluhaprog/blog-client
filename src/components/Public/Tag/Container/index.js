import {TagListContainer} from './TagListContainer';
export {TagListContainer};
