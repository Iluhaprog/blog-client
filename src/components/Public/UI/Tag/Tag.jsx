import React from 'react';
import styled from 'styled-components';
import * as PropTypes from 'prop-types';

const TagBox = styled.div`
  padding: ${({mod}) => {
    return mod === 'big' ? 10 : 5;
  }}px;
  
  border-radius: ${({mod}) => {
    return mod === 'big' ? 50 : 5;
  }}px;
  
  font-size: ${({mod}) => {
    return mod === 'big' ? 16 : 12;
  }}px;
  
  font-family: var(--public-font);
  color: #000;
  border: 1px solid #000;
  max-width: fit-content;
  
  margin: ${({mod}) => {
    return mod === 'big' ? 10 : 5;
  }}px;
`;

const Tag = ({title, mod}) => {
  return (
    <TagBox mod={mod}>
      {title}
    </TagBox>
  );
};

Tag.propTypes = {
  title: PropTypes.string,
  mod: PropTypes.oneOf(['big', 'small']),
};

export {Tag};
