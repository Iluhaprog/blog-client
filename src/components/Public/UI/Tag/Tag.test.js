import {render} from '@testing-library/react';
import {Tag} from './Tag';

test('Should render Tag component', () => {
  const {asFragment} = render(
    <Tag title={'TEST_TAG'} mod={'small'}/>
  );
  expect(asFragment()).toMatchSnapshot();
});
