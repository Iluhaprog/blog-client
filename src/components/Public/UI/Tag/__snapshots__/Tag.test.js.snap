// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Tag component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ gJkWUT"
  >
    TEST_TAG
  </div>
</DocumentFragment>
`;
